src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -78,82 +88,37 @@ func (a *StorageMinerActorCode_I) NotifyOfSurprisePoStChallenge(rt Runtime) Invo
 	a._expirePreCommittedSectors(rt)
 
 	h, st := a.State(rt)
-	st.ChallengeStatus().Impl().LastChallengeEpoch_ = rt.CurrEpoch()
+	// update challenge start epoch
+	st.ChallengeStatus().Impl().OnNewChallenge(rt.CurrEpoch())
 	UpdateRelease(rt, h, st)
 	return rt.SuccessReturn()
 }
 
-func (a *StorageMinerActorCode_I) _slashDealsFromFaultReport(rt Runtime, sectorNumbers []sector.SectorNumber, action deal.StorageDealSlashAction) {
-
-	h, st := a.State(rt)
-
-	dealIDs := make([]deal.DealID, 0)
-
-	for _, sectorNo := range sectorNumbers {
-
-		utilizationInfo := st._getUtilizationInfo(rt, sectorNo)
-		activeDealIDs := utilizationInfo.DealExpirationAMT().Impl().ActiveDealIDs()
-		dealIDs = append(dealIDs, activeDealIDs...)
-
-	}
-
-	Release(rt, h, st)
-
-	// TODO: Send(StorageMarketActor, ProcessDealSlash)
-}
-
-// construct FaultReport
-// reset NewTerminatedFaults
-func (a *StorageMinerActorCode_I) _submitFaultReport(
-	rt Runtime,
-	newDeclaredFaults sector.CompactSectorSet,
-	newDetectedFaults sector.CompactSectorSet,
-	newTerminatedFaults sector.CompactSectorSet,
-) {
-	faultReport := &sector.FaultReport_I{
-		NewDeclaredFaults_:   newDeclaredFaults,
-		NewDetectedFaults_:   newDetectedFaults,
-		NewTerminatedFaults_: newTerminatedFaults,
-	}
-
-	rt.Abort("TODO") // TODO: Send(SPA, ProcessFaultReport(faultReport))
-	panic(faultReport)
-
-	// only terminatedFault will be slashed
-	if len(newTerminatedFaults) > 0 {
-		a._slashDealsFromFaultReport(rt, newTerminatedFaults.SectorsOn(), deal.SlashTerminatedFaults)
-	}
-
-	h, st := a.State(rt)
-	st.SectorTable().Impl().TerminatedFaults_ = sector.CompactSectorSet(make([]byte, 0))
-	UpdateRelease(rt, h, st)
-}
-
-// construct PowerReport from SectorTable
-// need lastPoSt to search for new expired deals in _updateSectorUtilization
-// where DealExpirationAMT takes in a range of Epoch and return a list of values that expire in that range
-func (a *StorageMinerActorCode_I) _submitPowerReport(rt Runtime, lastPoSt block.ChainEpoch) {
-	h, st := a.State(rt)
-	newExpiredDealIDs := st._updateSectorUtilization(rt, lastPoSt)
-	activePower := st._getActivePower(rt)
-	inactivePower := st._getInactivePower(rt)
+// Called by the cron actor at every tick. Will return immediately if the miner
```

Comment:
I was under the impression that we don't need to poll every miner actor at every epoch - that's going to get expensive fast. Surely we only need to poll challenged actors, and it would be worth recording that set somewhere? @zixuanzh

Reply:
Also I recommend inserting an `OnCronTick` method, that may just immediately delegate to this one, but provide room for extending the functionality without causing the method name to drift.

Reply:
yes, every tick may not be every epoch @anorth (which is why i mucked around with how the state was stored to prevent miners from being able to submit PoSts due to cron delay even if they had passed deadlines.

The issue with the challenged set is that the current method of checking which are challenged uses onChainRandomness from the epoch to answer `ShouldChallenge()`. I think this is worth touching up, but in a different PR.